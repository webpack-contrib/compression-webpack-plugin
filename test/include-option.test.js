import Plugin from '../src/index';

import {
  cleanErrorStack,
  createCompiler,
  compile,
  getAssetsInfo,
} from './helpers';

describe('when applied with `include` option', () => {
  let compiler;

  beforeEach(() => {
    compiler = createCompiler({
      entry: {
        js: `${__dirname}/fixtures/entry.js`,
      },
      output: {
        path: `${__dirname}/dist`,
        filename: '[name].js?var=[hash]',
        chunkFilename: '[id].[name].js?ver=[hash]',
      },
    });
  });

  it('matches snapshot for a single `include` value ({RegExp})', () => {
    new Plugin({
      include: /\.js(\?.*)?$/i,
      minRatio: 1,
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });

  it('matches snapshot for multiple `include` values ({Array<RegExp>})', () => {
    new Plugin({
      include: [/\.js(\?.*)?$/i, /\.svg(\?.*)?$/i],
      minRatio: 1,
    }).apply(compiler);

    return compile(compiler).then((stats) => {
      const errors = stats.compilation.errors.map(cleanErrorStack);
      const warnings = stats.compilation.warnings.map(cleanErrorStack);

      expect(errors).toMatchSnapshot('errors');
      expect(warnings).toMatchSnapshot('warnings');
      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');
    });
  });
});
