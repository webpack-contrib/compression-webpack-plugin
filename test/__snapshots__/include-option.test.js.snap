// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`include\` option matches snapshot for a single \`include\` value ({RegExp}): assets 1`] = `
Array [
  Array [
    "1.async.js.gz?ver=1ea6f97e389bd951fd50",
    117,
  ],
  Array [
    "1.async.js?ver=1ea6f97e389bd951fd50",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "js.js.gz?var=1ea6f97e389bd951fd50",
    2773,
  ],
  Array [
    "js.js?var=1ea6f97e389bd951fd50",
    10201,
  ],
]
`;

exports[`when applied with \`include\` option matches snapshot for a single \`include\` value ({RegExp}): errors 1`] = `Array []`;

exports[`when applied with \`include\` option matches snapshot for a single \`include\` value ({RegExp}): warnings 1`] = `Array []`;

exports[`when applied with \`include\` option matches snapshot for multiple \`include\` values ({Array<RegExp>}): assets 1`] = `
Array [
  Array [
    "1.async.js.gz?ver=1ea6f97e389bd951fd50",
    117,
  ],
  Array [
    "1.async.js?ver=1ea6f97e389bd951fd50",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.js.gz?var=1ea6f97e389bd951fd50",
    2773,
  ],
  Array [
    "js.js?var=1ea6f97e389bd951fd50",
    10201,
  ],
]
`;

exports[`when applied with \`include\` option matches snapshot for multiple \`include\` values ({Array<RegExp>}): errors 1`] = `Array []`;

exports[`when applied with \`include\` option matches snapshot for multiple \`include\` values ({Array<RegExp>}): warnings 1`] = `Array []`;
